### DISCUSSION TITLE
Issues with my py2 backport of utils.named_pipe part 2 (closed, Python 2 is EOL)

### DISCUSSION THREAD

**Discussion by Billy2011** [streamlink/streamlink#3790] [Q&A] [answered]
Any idea why the posix test fails?
```
______________________ TestNamedPipePosix.test_named_pipe ______________________
self = <tests.test_utils_named_pipe.TestNamedPipePosix testMethod=test_named_pipe>
    def test_named_pipe(self):
        pipe = NamedPipePosix()
        self.assertTrue(pipe.path.is_fifo())
        reader = ReadNamedPipeThreadPosix(pipe)
        reader.start()
        pipe.open()
>       self.assertEqual(pipe.write(b"foo"), 3)
E       AssertionError: None != 3
```

**Comment by gravyboat** [comment, marked as answer]
Alright I'm putting my foot down here. Bastimeyer was kind enough to answer a few questions in the previous discussion but this is getting out of hand now. We do not support Python2. Supporting Python2, or helping people to support Python2 is out of scope for this project. If we wanted to deal with these issues we would still support Python2 which we do not and will not. There is no reason anyone should still be relying on Python2. Python3 is an install away on every maintained distro, it is time to upgrade. I'm marking this as out of scope and locking the discussion, please do not create any additional Python2 related issues or discussions. We are not supporting or helping people support [EOL versions of Python](https://www.python.org/doc/sunset-python-2/).
